### PR TITLE
#1521: VT100 escape codes DECSC and DECRC now handle character attributes

### PR DESCRIPTION
**Pull request by 7PH**
**Following [issue #1521](https://github.com/xtermjs/xterm.js/issues/1521)**

Here is the list of the modification I've made:
- added an attribute 'savedCurAttr' used to save  the 'curAttr' attribute of 'Terminal' and 'MockInputHandlingTerminal' 
- edited the saveCursor and restoreCursor of the InputHandler class to effectively store and restore the characters attributes
- added unit tests (and checked they passed)

I tried to stay as close as possible to your code style.

Please let me know if there is anything that requires action on my side.

### CODE REVIEW

**Comment by 7PH** [comment]
Hmm, i don't get why the **Windows Job / Initialize phase** is failing, I doubt my changes made that fail.

Let me know if there is anything I can do to help.

**Comment by Tyriar** [comment]
> Hmm, i don't get why the Windows Job / Initialize phase is failing

Yeah this wouldn't be related to your changes, it will probably pass next time.

File: src/InputHandler.ts [hunk 1]
```diff
@@ -1834,6 +1835,7 @@ export class InputHandler implements IInputHandler {
   public restoreCursor(params: number[]): void {
     this._terminal.buffer.x = this._terminal.buffer.savedX || 0;
     this._terminal.buffer.y = this._terminal.buffer.savedY || 0;
+    this._terminal.curAttr = this._terminal.savedCurAttr || 0;
```
Comment (Tyriar) [inline src/InputHandler.ts]: Should this be `|| DEFAULT_ATTR`?

**Comment by Tyriar** [comment]
Thanks a lot @7PH!

**Pull request merged**